library/core/src/sync/atomic.rs

```diff
@@ -107,6 +107,33 @@
 //! }
 //! ```
 //!
+//! Although the most convenient way to use an atomic type is to wrap it as an `Arc`,
+//! it is possible to share an atomic type between threads without `Arc`.
```

Comment:
Your PR description says:
> Currently, example of Atomic type like `AtomicUsize` are either under `Arc` or just a static variable, which either contains extra cost or has problem with thread safety.

But the actual documentation added in this commit doesn't mention this motivation of avoiding certain costs or thread safety problems. It probably should.